sdk/core/core-lro/samples/typescript/samplesClient.ts

```diff
@@ -0,0 +1,212 @@
+   */
+  previousResponses: ReturnValue[];
+}
+
+/**
+ * The definition of our sample's poll operation.
+ * It extends PollOperation, which makes sure the shape of the interface remains consistent.
+ * It uses the SamplePollOperationState, which allows to use the default properties of a poll operation state,
+ * plus the ones we've defined above.
+ * It returns an instance of the ReturnValue interface.
+ */
+interface SamplePollOperation extends PollOperation<SamplePollOperationState, ReturnValue> {}
+
+/**
+ * A utility method that builds an instance of the sample poll operation with some default methods.
+ */
+function makeSamplePollOperation(
+  state: SamplePollOperationState
+): SamplePollOperation {
+  return {
+    /**
+     * To ensure the state is always clean of previously used references, we're making a copy of it.
+     */
+    state: {
+      ...state
+    },
+    /**
+     * The update method should change the operation's state.
+     * In our case, we'll be calling the Client's makeRequest.
+     */
+    async update(): Promise<SamplePollOperation> {
+      const state = this.state;
+
+      if (!state.isStarted) {
+        state.result = await state.client.makeRequest();
+        state.isStarted = true;
+      } else
+      if (!state.isCompleted) {
+        state.result = await state.client.makeRequest();
+        if (state.client.isDone(state.result)) {
+          state.isCompleted = true;
+        }
+      }
+
+      state.previousResponses.push(state.result);
+    
+      return makeSamplePollOperation(state);
+    },
+    /**
+     * The cancel method can be called to send a cancelling signal to the remote service.
+     * If cancelling isn't supported, an error can be thrown instead.
+     */
+    async cancel(): Promise<SamplePollOperation> {
+      const state = this.state;
+
+      state.result = await state.client.cancel();
+
+      state.previousResponses.push(state.result);
+    
+      return makeSamplePollOperation({
+        ...state,
+        isCancelled: true
+      });
+    },
+
+    /**
+     * The toString method serializes the PollOperationState, so that
+     * one poller can be stopped and another poller can recover at any time
+     * based on a copy of the first one.
+     */
+    toString(): string {
+      return JSON.stringify({
+        state: this.state
+      });
+    }
+  };
+}
+
+/**
+ * Here's our sample's poller.
+ * It will work over our SamplePollOperationState.
+ * Once the polling is completed, it will return an instance of the ReturnValue interface.
+ */
+class SamplePoller extends Poller<SamplePollOperationState, ReturnValue> {
+
+  public intervalInMs: number;
+
+  constructor({
+    client,
+    intervalInMs = 2000,
+    resumeFrom,
+  }: {
+    client: Client;
+    intervalInMs?: number;
+    resumeFrom?: string;    
+  }) {
+
+    let state: SamplePollOperationState | undefined;
```

Comment:
it feels kinda weird that we don't initialize the boolean properties of state (e.g. `isStarted`) to `false` and instead let them be optional and rely on `undefined` being falsy. It's probably not worth the hassle to bother initializing them, but it just felt a little odd to me.

Reply:
We haven't been initializing any of these properties in any of the implementations yet, I think. We assume that they're false if they're not true. Let me know if we should change it.